sunpy/coordinates/screens.py

```diff
@@ -39,6 +42,73 @@ def __exit__(self, exc_type, exc_value, exc_tb):
+        typically take no more than 6 iterations. When working with an array coordinate,
+        elements that have converged are skipped in subsequent iterations.
+        """
+        # Check if the logging level is at least DEBUG (for performance reasons)
+        debug_output = log.getEffectiveLevel() <= logging.DEBUG
+
+        log.debug("Differentially rotating the screen")
+        use = np.ones(distance.shape, dtype=bool)
+        delta = np.empty_like(distance)
+        next_distance = np.empty_like(distance)
+        for niter in range(20):
+            # If this is not the first iteration, update best distance and calculate new distance
+            if niter > 0:
+                # Calculate for only those pixels that do not meet the tolerance
+                with np.errstate(invalid='ignore'):
+                    use = np.abs(delta / distance) >= 1e-11*u.one
+                if np.sum(use) == 0:
+                    log.debug(f"Solved for the differentially rotated screen after {niter} iterations")
+                    break
+                # Squash small deltas to avoid warnings from later expressions
+                delta = np.where(use, delta, 0)
+
+                if niter == 1:
+                    # For the second iteration, use a simple guess
+                    best_distance, distance = distance, distance + delta
+                    best_delta = delta.copy()
+                else:
+                    # Starting with the third iteration, linearly interpolate between the latest distance and the best distance
+                    with np.errstate(invalid='ignore'):
+                        next_distance[use] = distance[use] - delta[use] * np.nan_to_num((best_distance[use] - distance[use]) / (best_delta[use] - delta[use]))
+
+                    # Update the best distance (not including the most recent calculation)
+                    best_distance = np.where(np.abs(best_delta) < np.abs(delta), best_distance, distance)
+                    best_delta = np.where(np.abs(best_delta) < np.abs(delta), best_delta, delta)
+                    distance = next_distance.copy()
+
+            # Calculate corresponding new delta
+            other_3d = coord[use].realize_frame(coord[use].represent_as(UnitSphericalRepresentation) * distance[use])
+            native_3d = other_3d.transform_to(screen_frame)
+            native_2d = native_3d.realize_frame(native_3d.represent_as(UnitSphericalRepresentation))
+            delta[use] = self.calculate_distance(native_2d) - native_3d.spherical.distance
+
+            if debug_output:
+                log.debug(f"npoints={np.sum(use)}; max distance={distance.max()}, max delta={delta.max()}"
+                          + (f", best distance={best_distance.max()}, best delta={best_delta.max()}" if niter > 0 else ""))
+        else:
+            warn_user(f"Failed to solve for the differentially rotated screen after {niter+1} iterations")
```

Comment:
Would it be worth reporting the error when the max number of iterations was reached? If things fail, this would at least give the user an idea as to how bad things have gone.

Reply:
It's a good suggestion, but I'm hesitant to report a value here because I can't report it in the "right" frame...  A power user can turn on debug logging and sift through the crazy amount of messages.